js/src/components/google-combo-account-card/claim-ads-account/claim-ads-account.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Flex, FlexBlock } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import ClaimAdsAccountButton from './claim-ads-account-button';
+import Section from '.~/wcdl/section';
+import useWindowFocus from '.~/hooks/useWindowFocus';
+import useGoogleAdsAccountStatus from '.~/hooks/useGoogleAdsAccountStatus';
+import { useAppDispatch } from '.~/data';
+import './claim-ads-account.scss';
+import useUpsertAdsAccount from '.~/hooks/useUpsertAdsAccount';
+
+/**
+ * ClaimAdsAccount component.
+ *
+ * @return {JSX.Element} ClaimAdsAccount component.
+ */
+const ClaimAdsAccount = () => {
+	const [ , forceUpdate ] = useState( 0 );
+	const isWindowFocused = useWindowFocus();
+	const { hasAccess, hasFinishedResolution, inviteLink, step } =
+		useGoogleAdsAccountStatus();
+	const [ upsertAdsAccount, { loading } ] = useUpsertAdsAccount();
+	const { invalidateResolution } = useAppDispatch();
+	const claimButtonClickedRef = useRef( false );
+	const claimInProgressRef = useRef( false );
+
+	useEffect( () => {
+		if ( isWindowFocused && claimButtonClickedRef.current ) {
+			invalidateResolution( 'getGoogleAdsAccountStatus' );
+			forceUpdate( ( prev ) => prev + 1 );
+		}
+	}, [ isWindowFocused, invalidateResolution ] );
+
+	useEffect( () => {
+		const handlePostClaimActions = async () => {
+			await upsertAdsAccount();
+			claimInProgressRef.current = false;
+		};
+
+		if (
+			hasAccess === true &&
+			step === 'conversion_action' &&
+			! claimInProgressRef.current
+		) {
+			claimInProgressRef.current = true;
```

Comment:
@ankitrox I don't think we should check whether there's a claim in progress to upsert an Ads account.  I think `hasAccess` and `step === 'conversion_action'` should be enough:
useEffect( () => {
		if ( hasAccess === true && step === 'conversion_action' ) {
			upsertAdsAccount();
		}
	}, [ hasAccess, upsertAdsAccount, step ] );

Reply:
This has been changed to

```js
useEffect( () => {
if ( hasAccess === true && step === 'conversion_action' ) {
	upsertAdsAccount();
}
}, [ hasAccess, step, upsertAdsAccount ] );
```